### PR TITLE
Dev troubleshooting

### PR DESCRIPTION
**Pull request by ngaskill**
Contains troubleshooting articles for:
- LRDOCS-3878
- LRDOCS-3880
- LRDOCS-3881
- LRDOCS-3882
- LRDOCS-3883
- LRDOCS-3917
- LRDOCS-3919
- LRDOCS-3920

@jhinkey

### CODE REVIEW

File: develop/tutorials/articles/310-troubleshooting/04-jsp-fragments.markdown [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Why Aren't JSP overrides I Made Using Fragments Showing? [](id=why-arent-jsp-overrides-i-made-using-fragments-showing)
+
+The fragment module must specify the exact version of the host module. A 
+@product@ upgrade might have changed some JSPs in the host module, prompting a 
+version update. If this occurs, check that your JSP customizations are 
+compatible with the updated host JSPs and then update your fragment module's 
+targeted version to match the host module. For more information, 
+[click here](/develop/tutorials/-/knowledge_base/7-0/overriding-a-modules-jsps) 
+to see the tutorial on overriding app JSPs. 
+
```
Comment (sez11a) [inline develop/tutorials/articles/310-troubleshooting/04-jsp-fragments.markdown]: You should have a Bnd.bnd example here.

Comment (jrao) [reply develop/tutorials/articles/310-troubleshooting/04-jsp-fragments.markdown]: Here's the contents of the `bnd.bnd` file from my example with the fragment host specified:

```
Bundle-Name: custom-login-jsp
Bundle-SymbolicName: custom.login.jsp
Bundle-Version: 1.0.0
Fragment-Host: com.liferay.login.web;bundle-version="1.1.18"
```

Full example project is here: https://github.com/jrao/7.0.3-ga4-liferay-workspace/tree/master/modules/custom-login-jsp

Comment (ngaskill) [reply develop/tutorials/articles/310-troubleshooting/04-jsp-fragments.markdown]: Thanks guys. @sez11a I added the bnd example and a link to a tutorial on configuring dependencies, which contains a section on finding deployed module versions.

**Comment by ngaskill** [comment]
@sez11a I deleted the IDE debugging article, so these are ready for review.

**Pull request closed without merging**